services/src/main/java/org/fao/geonet/api/pages/PagesAPI.java

```diff
@@ -407,6 +424,25 @@ public ResponseEntity<List<org.fao.geonet.api.pages.PageProperties>> listPages(
                         filteredResult.add(new org.fao.geonet.api.pages.PageProperties(page));
                     }
                 }
+            } else if (page.getStatus().equals(Page.PageStatus.PRIVATE_GROUP) && us.getProfile() != null && us.getProfile() != Profile.Guest) {
+                String myUserId = us.getUserId();
+                String group = page.getAccessExpression();
```

Comment:
This variable is only used in the else block

`} else if (StringUtils.isNotEmpty(myUserId)) {`

So maybe the variable should only be initialized in that block.
Otherwise, it will set the value needlessly for administrator users.

Reply:
Not sure about this comment. The variable is needed to initialized before the else if block. I have renamed the variable to currentUserId